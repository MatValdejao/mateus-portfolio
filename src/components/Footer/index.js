import React from "react";

function Footer() {
	return (
		<footer>
			<address className="flex-row">
				<div className="mx-2">
					<a
						data-testid="github"
						rel="noreferrer"
						href="https://github.com/MatValdejao"
						target="_blank"
					>
						GitHub
					</a>
				</div>
				<div className="mx-2">
					<a
						data-testid="linkedin"
						rel="noreferrer"
						href="https://www.linkedin.com/in/mateusvaldejao/"
						target="_blank"
					>
						LinkedIn
					</a>
				</div>
				<div className="mx-2">
					<a
						data-testid="instagram"
						rel="noreferrer"
						href="https://www.instagram.com/mateusvaldejao/"
						target="_blank"
					>
						Instagram
					</a>
				</div>
			</address>
		</footer>
	);
}

export default Footer;
