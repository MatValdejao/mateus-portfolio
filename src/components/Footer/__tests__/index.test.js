import React from "react";
import { render, cleanup, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Footer from "..";

afterEach(cleanup)

describe("footer component", () => {
    it("renders", () => {
        render(<Footer/>)
    })

    it("matches snapshot", () => {
        const { asFragment } = render(<Footer />)
        
        expect(asFragment()).toMatchSnapshot()
    })
})


it("links are visible", () => {
    render(<Footer />)
    
    expect(screen.getByTestId("github")).toHaveTextContent("GitHub")
    expect(screen.getByTestId("linkedin")).toHaveTextContent("LinkedIn")
    expect(screen.getByTestId("instagram")).toHaveTextContent("Instagram")
})
