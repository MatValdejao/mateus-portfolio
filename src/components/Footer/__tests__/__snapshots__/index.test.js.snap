// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer component matches snapshot 1`] = `
<DocumentFragment>
  <footer>
    <address
      class="flex-row"
    >
      <div
        class="mx-2"
      >
        <a
          data-testid="github"
          href="https://github.com/MatValdejao"
          rel="noreferrer"
          target="_blank"
        >
          GitHub
        </a>
      </div>
      <div
        class="mx-2"
      >
        <a
          data-testid="linkedin"
          href="https://www.linkedin.com/in/mateusvaldejao/"
          rel="noreferrer"
          target="_blank"
        >
          LinkedIn
        </a>
      </div>
      <div
        class="mx-2"
      >
        <a
          data-testid="instagram"
          href="https://www.instagram.com/mateusvaldejao/"
          rel="noreferrer"
          target="_blank"
        >
          Instagram
        </a>
      </div>
    </address>
  </footer>
</DocumentFragment>
`;
